src/libstd/io/stdio.rs

```diff
@@ -792,10 +792,14 @@ fn print_to<T>(
 {
     let result = local_s
         .try_with(|s| {
-            if let Ok(mut borrowed) = s.try_borrow_mut() {
-                if let Some(w) = borrowed.as_mut() {
-                    return w.write_fmt(args);
-                }
+            // Note that we completely remove a local sink to write to in case
+            // our printing recursively panics/prints, so the recursive
+            // panic/print goes to the global sink instead of our local sink.
+            let prev = s.borrow_mut().take();
+            if let Some(mut w) = prev {
+                let result = w.write_fmt(args);
+                *s.borrow_mut() = Some(w);
+                return result;
```

Comment:
Just to make sure I'm understanding correctly, the reason this does not need a try_borrow_mut anymore is because there's no possibility of a recursive lock, right? I believe the try_borrow was added in https://github.com/rust-lang/rust/commit/e2fd2dffde52a59f7d59d67460aeb2ebf33f77dd FWIW.

Reply:
Right yeah the scope of the borrow should always be one line of code which doesn't execute other arbitrary lines of code (e.g. an arbitrary `Write` implementation). As a result we should be guaranteed that it always succeeds.